src/lib/packet.h

```diff
@@ -565,18 +572,29 @@ typedef enum {
 
 #define PGP_SA_DEFAULT_CIPHER PGP_SA_CAST5
 
+#define PGP_SA_DEFAULT_CIPHER_MODE PGP_CIPHER_MODE_CFB
+
 void pgp_calc_mdc_hash(
   const uint8_t *, const size_t, const uint8_t *, const unsigned, uint8_t *);
 unsigned pgp_is_hash_alg_supported(const pgp_hash_alg_t *);
 
+#define PGP_PROTECTED_AT_SIZE 15
+
+typedef struct pgp_key_t pgp_key_t;
+
+typedef struct pgp_seckey_t pgp_seckey_t;
+
+typedef pgp_seckey_t *pgp_seckey_decrypt_t(const pgp_key_t *key, FILE *passfp);
+
 /** pgp_seckey_t
  */
-typedef struct pgp_seckey_t {
```

Comment:
I think it's better to keep the struct and typedef together as it's a pretty strong convention in the code I see. You can always pre-declare the struct and use something like this:

```diff
diff --git a/src/lib/packet.h b/src/lib/packet.h
index 095996d..5b23279 100644
--- a/src/lib/packet.h
+++ b/src/lib/packet.h
@@ -582,13 +582,12 @@ unsigned pgp_is_hash_alg_supported(const pgp_hash_alg_t *);
 
 typedef struct pgp_key_t pgp_key_t;
 
-typedef struct pgp_seckey_t pgp_seckey_t;
-
-typedef pgp_seckey_t *pgp_seckey_decrypt_t(const pgp_key_t *key, FILE *passfp);
+struct pgp_seckey_t;
+typedef struct pgp_seckey_t *pgp_seckey_decrypt_t(const pgp_key_t *key, FILE *passfp);
 
 /** pgp_seckey_t
  */
-struct pgp_seckey_t {
+typedef struct pgp_seckey_t {
     pgp_pubkey_t        pubkey; /* public key */
     pgp_s2k_usage_t     s2k_usage;
     pgp_s2k_specifier_t s2k_specifier;
@@ -611,7 +610,7 @@ struct pgp_seckey_t {
     uint8_t *             encrypted;
     pgp_seckey_decrypt_t *decrypt_cb;
     const char *protected_at[PGP_PROTECTED_AT_SIZE + 1]; // keep 1 byte for \0 and padding
-};
+} pgp_seckey_t;
 
 /** Signature Type.
  * OpenPGP defines different signature types that allow giving
```